File: android/src/main/kotlin/tech/shmy/a_player/player/APlayerListener.kt
package tech.shmy.a_player.player

interface APlayerListener {
    fun onPlayingListener(isPlaying: Boolean)
    fun onErrorListener(code: String, message: String)
    fun onCompletionListener()
    fun onReadyToPlayListener()
    fun onVideoSizeChangedListener(width: Int, height: Int)
    fun onCurrentPositionChangedListener(position: Long)
    fun onCurrentDownloadSpeedChangedListener(speed: Long)
    fun onBufferedPositionChangedListener(buffered: Long)
    fun onLoadingBeginListener()
    fun onLoadingProgressListener(percent: Int)
    fun onLoadingEndListener()
}

File: ios/Classes/Player/APlayerListener.swift
//
//  APlayerListener.swift
//  a_player
//
//  Created by drm-chao wang on 2022/7/4.
//

import Foundation

protocol APlayerListener {
    func onInitializedListener() -> Void
    func onPlayingListener(isPlaying: Bool) -> Void
    func onErrorListener(code: String, message: String) -> Void
    func onCompletionListener() -> Void
    func onReadyToPlayListener() -> Void
    func onVideoSizeChangedListener(width: Int32, height: Int32) -> Void
    func onCurrentPositionChangedListener(position: Int64) -> Void
    func onCurrentDownloadSpeedChangedListener(speed: Int64) -> Void
    func onBufferedPositionChangedListener(buffered: Int64) -> Void
    func onSwitchToSoftwareVideoDecoderListener() -> Void
    func onLoadingBeginListener() -> Void
    func onLoadingProgressListener(percent: Int) -> Void
    func onLoadingEndListener() -> Void
    
}
